### ISSUE TITLE
Invalid URLs (absolute paths) in source maps in some cases

### ISSUE THREAD

**Issue by RReverser** [emscripten-core/emscripten#9837]
## Summary

Source maps are supposed to contain only valid URLs in the `sources` array (either relative, or absolute with scheme). Emscripten sometimes emits absolute paths into that array instead, making them impossible to load by a browser.

## Examples

For a file `C:\Users\rreverser\Projects\temp-wasm\temp.c` with arbitary content in a folder `temp-wasm`, I'm trying different formats for the filename.

Note that this is not a Windows-specific issue, and you'll get same results with Unix filepaths.

1. Running `emcc ./temp.c -g4 -o temp.wasm` (relative filename in the same folder):

```js
{"version":3,"sources":["./temp.c"],"names":[],...
```

2. Running `emcc temp.c -g4 -o temp.wasm` (same relative filename, but without explicit `./`):

```js
{"version":3,"sources":["C:/Users/rreverser/Projects/temp-wasm/temp.c"],"names":[],...
```

3. Running `emcc ../temp-wasm/temp.c` (relative filename with a folder):

```js
{"version":3,"sources":["../temp-wasm/temp.c"],"names":[],
```

4. Running `emcc C:\Users\rreverser\Projects\temp-wasm\temp.c -g4 -o temp.wasm` (absolute filename):

```js
{"version":3,"sources":["C:/Users/rreverser/Projects/temp-wasm/temp.c"],"names":[],...
```

## Results

When a filename is relative but without a leading `./` or is specified as absolute, Emscripten outputs an absolute filename in `sources`, which is not a valid URL and can't be handled by DevTools. It could be valid if prefixed with `file:///` specifier, but that's not an ideal solution.

The only case where Emscripten `sources` ends up being valid URL is when a filename is specified as explicitly relative (cases 1 and 3).

Then, Emscripten outputs a relative URL as well, which can be handled well by DevTools both in case of loading from filesystem as well as a static HTTP server.

More so, even though case 3 outputs a relative filename, it's not ideal.

## Suggested solution

Emscripten should always compute relative URL out of the source map filename and the actual sources, and emit that in `sources` array.

**Comment by kripken** [comment]
Sorry, I may be missing something here, the output in cases 1, 2, and 4 look identical? They are all
```javascript
{"version":3,"sources":["C:/Users/rreverser/Projects/temp-wasm/temp.c"],"names":[],...
```
All those are absolute paths, I think? That seems to conflict with 

> The only case where Emscripten sources ends up being valid URL is when a filename is specified as explicitly relative (cases 1 and 3).

**Comment by kripken** [comment]
Maybe related to https://github.com/emscripten-core/emscripten/pull/7481/files ?

**Comment by RReverser** [comment]
> Sorry, I may be missing something here, the output in cases 1, 2, and 4 look identical?

Oops, sorry - copy-paste mistake. Fixed output of the case 1.

**Comment by RReverser** [comment]
> Maybe related to emscripten-core/emscripten/pull/7481/files ?

Maybe it was revealed by it, but definitely not the original cause as the change in PR seems correct.

**Comment by kripken** [comment]
Got it, thanks @RReverser! Fix in #9882, PTAL
